stl/inc/tuple

```diff
@@ -1079,23 +1079,39 @@ constexpr decltype(auto) apply(_Callable&& _Obj, _Tuple&& _Tpl) noexcept(
         make_index_sequence<tuple_size_v<remove_reference_t<_Tuple>>>{});
 }
 
+template <class _Ty, class _Tuple, class _Seq = make_index_sequence<tuple_size_v<remove_reference_t<_Tuple>>>,
+    class = void>
+inline constexpr bool _Can_make_from_tuple = false;
+template <class _Ty, class _Tuple, size_t... _Indices>
+inline constexpr bool _Can_make_from_tuple<_Ty, _Tuple, index_sequence<_Indices...>,
+    enable_if_t<is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD declval<_Tuple>()))...>>> = true;

```

Comment:
This layer of `enable_if_t` is unnecessary. We know that `_Tuple` is *`tuple-like`*, and we've formed the `_Indices` ourselves. Therefore, the `get<_Indices>` will be well-formed, so we don't need to defend against not being able to form the arguments to `is_constructible_v`.

Reply:
However, `make_from_tuple`/_`make-from-tuple-impl`_ didn't reject non-tuple-like types by SFINAE in C++17/20 (the constraints were added in WG21-P2165R4), so the `get` calls might result in hard error.

BTW, one could write a class publicly derived from a tuple-like class with `tuple_size`/`tuple_element` correctly specialized for it. `apply` and `make_from_tuple` used to work with such a class in C++17/20.

Reply:
I was wonder if do we need to reserve this for non-tuple-like types in C++17/C++20?